File: server/model/establishment.go
package model

// Establishment represents a business establishment
type Establishment struct {
	ID            int64  `json:"id"`
	Number        string `json:"number"`
	Name          string `json:"name"`
	CorporateName string `json:"corporate_name"`
	Address       string `json:"address"`
	City          string `json:"city"`
	State         string `json:"state"`
	ZipCode       string `json:"zip_code"`
	AddressNumber string `json:"address_number"`
}

File: web/src/types/establishment.ts
import type { StoreWithEstablishment } from "~/types/store"

export interface Establishment {
  id: string
  number: string
  name: string
  corporate_name: string
  address: string
  address_number: string
  city: string
  state: string
  zip_code: string
}

export interface EstablishmentWithStores extends Establishment {
  stores: StoreWithEstablishment[]
}
